lib/puppet/functions/empty.rb

```diff
@@ -27,7 +27,7 @@
   end
 
   dispatch :string_empty do
-    param 'String', :str
+    param 'Variant[String, Sensitive[String]]', :str
```

Comment:
I'm concerned about opening this can of worms. If we allow this, then what about camelcase, chomp, compare, downcase, length, lstrip, etc...

Reply:
Hi, @joshcooper, with all Respect for your Concearns, but it is unlikely that someone would want to run `camelcase` on a Password – the other Examples alike.   

But f.e. coming from [this ERB-File](https://github.com/puppetlabs/puppetlabs-haproxy/blob/main/templates/haproxy_userlist_block.erb) I definitely need the Function `empty` without having to `unwrap` the Arguments in [this EPP-File](https://github.com/cocker-cc/puppetlabs-haproxy/blob/Use_Puppet-Datatype_Sensitive/templates/haproxy_userlist_block.epp).  Concearning `Sensitive` an EPP is superiour to an ERB, because there never is a Need to `unwrap` – it just works due to [this Function](https://github.com/puppetlabs/puppet/blob/main/lib/puppet/functions/epp.rb).

Reply:
Yeah camelcase is bad example, I'm more concerned about `lstrip`, `rstrip`, `length` and `size`.

That said I can understand how checking for `if var && !var.empty?` is a common pattern, so I'm :+1: with this. One caveat, it'd be good to create a separate dispatch for `sensitive_empty` instead of overloading `string_empty`.

Reply:
Done